src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -302,19 +318,33 @@ private void collectionRoutes(){
                         phoneNumber = ((JSONObject) payer.get(1)).getString("value");
                         accountId = ((JSONObject) payer.get(0)).getString("value");
                     }
+                    if(amsIdentifier.equals(amsUtils.ACCOUNTID)){
+                        //go for mpesa workflow
+                         tenantSpecificBpmn = mpesaFlow.replace("{dfspid}", tenantId).replace("{ams}",amsUtils.AMS_SECONDARY);
+                    }
+
+                    else if(amsIdentifier.equals(amsUtils.FOUNDATIONALID)){
+                        //paygops workflow
+
+                        tenantSpecificBpmn = mpesaFlow.replace("{dfspid}", tenantId).replace("{ams}",amsUtils.AMS_DEFAULT);
+                    }
+                    else{
+                        //default paygops
+                        accountId = "0";
+                        tenantSpecificBpmn = mpesaFlow.replace("{dfspid}", tenantId).replace("{ams}",amsUtils.AMS_DEFAULT);
+
+                    }
+
                     String amount = body.getJSONObject("amount").getString("amount");
 
-                    extraVariables.put("accountId", accountId);
+                    extraVariables.put("accountId", accountId);//will have memoid in case of paygops
```

Comment:
No hardcoded logic around identifiers or AMS should be there in channel connector.